m5stick-c/arduino/sample/sample.ino

```diff
@@ -0,0 +1,163 @@
+#include <M5StickC.h>
+#include <BLEServer.h>
+#include <BLEDevice.h>
+#include <BLEUtils.h>
+#include <BLE2902.h>
+
+// Device Name: Maximum 30 bytes
+#define DEVICE_NAME "LINE Things Trial M5StickC"
+
+// User service UUID: Change this to your generated service UUID
+#define USER_SERVICE_UUID "91E4E176-D0B9-464D-9FE4-52EE3E9F1552"
+// User service characteristics
+#define WRITE_CHARACTERISTIC_UUID "E9062E71-9E62-4BC6-B0D3-35CDCD9B027B"
+#define NOTIFY_CHARACTERISTIC_UUID "62FBD229-6EDD-4D1A-B554-5C4E1BB29169"
+
+// PSDI Service UUID: Fixed value for Developer Trial
+#define PSDI_SERVICE_UUID "E625601E-9E55-4597-A598-76018A0D293D"
+#define PSDI_CHARACTERISTIC_UUID "26E2B12B-85F0-4F3F-9FDD-91D114270E6E"
+
+BLEServer* thingsServer;
+BLESecurity *thingsSecurity;
+BLEService* userService;
+BLEService* psdiService;
+BLECharacteristic* psdiCharacteristic;
+BLECharacteristic* writeCharacteristic;
+BLECharacteristic* notifyCharacteristic;
+
+bool deviceConnected = false;
+bool oldDeviceConnected = false;
+
+class serverCallbacks: public BLEServerCallbacks {
+  void onConnect(BLEServer* pServer) {
+   deviceConnected = true;
+  };
+
+  void onDisconnect(BLEServer* pServer) {
+    deviceConnected = false;
+  }
+};
+
+class writeCallback: public BLECharacteristicCallbacks {
+  void onWrite(BLECharacteristic *bleWriteCharacteristic) {
+    std::string value = bleWriteCharacteristic->getValue();
+    if ((char)value[0] <= 1) {
+      if ((char)value[0] == 1) {
+        digitalWrite(M5_LED, LOW);
+      }
+      else {
+        digitalWrite(M5_LED, HIGH);
+      }
+    }
+  }
+};
+
+void setup() {
+  Serial.begin(115200);
+  BLEDevice::init("");
+  BLEDevice::setEncryptionLevel(ESP_BLE_SEC_ENCRYPT_NO_MITM);
+
+  // Security Settings
+  BLESecurity *thingsSecurity = new BLESecurity();
+  thingsSecurity->setAuthenticationMode(ESP_LE_AUTH_REQ_SC_ONLY);
+  thingsSecurity->setCapability(ESP_IO_CAP_NONE);
+  thingsSecurity->setInitEncryptionKey(ESP_BLE_ENC_KEY_MASK | ESP_BLE_ID_KEY_MASK);
+
+  setupServices();
+  startAdvertising();
+
+  pinMode(M5_BUTTON_HOME, INPUT);
+
+  pinMode(M5_LED, OUTPUT);
+  digitalWrite(M5_LED, HIGH);
+
+  // M5StickC LCD Setup
+  M5.begin(true, true, false);
+  uint8_t brightness = 8;
+  M5.Axp.ScreenBreath(brightness);
+  M5.Lcd.qrcode("line://nv/things/deviceLink", 0, 0, 80, 9);
+  M5.Lcd.setTextColor(YELLOW);
+  M5.Lcd.setCursor(0, 80, 2);
+  M5.Lcd.println("Not Connect");
+  Serial.println("Not Connect");
+}
+
+uint8_t btnValue = 0;
+
+void loop() {
+
+  if (digitalRead(M5_BUTTON_HOME) == LOW && btnValue == 0) {
```

Comment:
I think this logic does not work well when button is chattering. (many notification will be sent)
Could you use button API instead of `digitalRead()` like M5Stack code?
https://github.com/line/line-things-starter/blob/master/m5stack/arduino/sample/sample.ino#L93
https://github.com/m5stack/M5StickC/blob/master/src/utility/Button.h#L24